openhtf/util/logs.py

```diff
@@ -206,24 +242,23 @@ def handle(self, record):
   def emit(self, record):
     """Save a logging.LogRecord to our test record.
 
-    LogRecords carry a significant amount of information with them including the
-    logger name and level information.  This allows us to be a little clever
-    about what we store so that filtering can occur on the client.
+    Logs carry useful metadata such as the logger name and level information.
+    We capture this in a structured format in the test record to enable
+    filtering by client applications.
 
     Args:
-      record: A logging.LogRecord to log.
+      record: A logging.LogRecord to record.
     """
-    message = record.getMessage()
-    if record.exc_info:
-      message += '\n' + ''.join(traceback.format_exception(
-          *record.exc_info))
-
-    log_record = LogRecord(
-        record.levelno, record.name, os.path.basename(record.pathname),
-        record.lineno, int(record.created * 1000), message
-    )
-    self._test_record.log_records.append(log_record)
-    self._notify_update()
+    try:
+      message = self.format(record)
```

Comment:
Using `self.format(record)` to use the built in traceback handling.

Before:

```
{
  "timestamp_millis": 1526597056909, 
  "level": 40, 
  "source": "hello_world.py", 
  "lineno": 62, 
  "message": "Hello World!\nTraceback (most recent call last):\n  File \"/Users/kschiller/open/openhtf/examples/hello_world.py\", line 60, in hello_world\n    raise RuntimeError\nRuntimeError\n", 
  "logger_name": "openhtf.test_record.29880:8207c3bec1ba458c:17be7a1147a440c0:1526597055785.phase.hello_world"
}
```

After:

```
{
  "timestamp_millis": 1526597049554, 
  "level": 40, 
  "source": "hello_world.py", 
  "lineno": 62, 
  "message": "Hello World!\nTraceback (most recent call last):\n  File \"/Users/kschiller/open/openhtf/examples/hello_world.py\", line 60, in hello_world\n    raise RuntimeError\nRuntimeError", 
  "logger_name": "openhtf.test_record.29830:f0b1de88e7a24826:1caf62674ed14239:1526597048482.phase.hello_world"
}
```